src/uwtools/drivers/jedi.py

```diff
@@ -0,0 +1,134 @@
+
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file, filecopy, symlink
+
+
+class JEDI(Driver):
+    """
+    A driver for the JEDI component.
+    """
+
+    def __init__(self, config: Path, cycle: datetime, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.files_copied(),
+            self.files_linked(),
+            self.configuration_file(),
+            self.runscript(),
+        ]
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
+
+    @task
+    def validate_only(self):
+        """
+        Validate config.
+        """
+        taskname = self._taskname("validate_only")
+        yield taskname
+        a = asset(None, lambda: False)
+        yield a
+        path = Path(self._driver_config["configuration_file"]["base_file"])
+        executable = Path(self._driver_config["execution"]["executable"])
+        yield [file(executable), file(path)]
+        env = " && ".join(self._driver_config["execution"]["envcmds"])
+        cmd = "{env} && time {x} --validate-only {p} 2>&1".format(env=env, x=executable, p=path)
+        result = run(taskname, cmd)
+        if result.success:
+            logging.info("%s: Config is valid", taskname)
+            a.ready = lambda: True
+
+    @task
+    def configuration_file(self):
+        """
+        The configuration file.
+        """
+        fn = "input.yaml"
```

Comment:
Could we name this one `config.yaml`? It's more consistent with the file name JCSDA uses in the [JEDI documentation](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/using/building_and_running/config_content.html) and `input.nml` is a very FV3-specific name.

Another option is `jedi.yaml`.

Yet another...we could use the name of the file provided as the base file. (Not sure I love this one given we're modifying it)

Reply:
I agree, I think picking either config.yaml or jedi.yaml, and using that across the board will keep it consistent. Since JCSDAs documentation uses config.yaml. I think I’ll go with that one.